src/Chapter3.hs

```diff
@@ -1052,19 +1053,13 @@ implement the following functions:
 -}
 
 data Weekday = Sunday | Monday | Tuesday | Wednesday | Thursday | Friday | Saturday
-  deriving (Show, Read, Eq, Ord)
+  deriving (Show, Read, Eq, Ord, Enum)
 
 isWeekend:: Weekday -> Bool 
 isWeekend day = day == Sunday || day == Saturday
 
 nextDay:: Weekday -> Weekday
-nextDay Sunday = Monday
-nextDay Monday = Tuesday
-nextDay Tuesday = Wednesday
-nextDay Wednesday = Thursday
-nextDay Thursday = Friday
-nextDay Friday = Saturday
-nextDay Saturday = Sunday
+nextDay = succ
```

Comment:
Unfortunately, `succ` is a partial function and it will fail in runtime on `Saturday` 😞

Reply:
thankyou

wish there was some generic marker for partial functions